### PR TITLE
SC-905 : In private Update user profile allow user to update org-association :Test cases added 

### PR DESCRIPTION
**Pull request by iostream04**

### CODE REVIEW

**Comment by iostream04** [comment]
@amolnin

File: service/src/test/java/org/sunbird/learner/actors/UserManagementActorTest.java [hunk 1]
```diff
@@ -358,6 +365,173 @@ public void testUpdateUserFailureWithUserTypeTeacher() {
     assertTrue(result);
   }
 
+  @Test
+  public void testUpdateUserPrivateFailureWithoutUserId() {
```
Comment (amolnin) [inline service/src/test/java/org/sunbird/learner/actors/UserManagementActorTest.java]: testUpdateUserOrgFailureWithoutUserId

File: service/src/test/java/org/sunbird/learner/actors/UserManagementActorTest.java [hunk 2]
```diff
@@ -358,6 +365,173 @@ public void testUpdateUserFailureWithUserTypeTeacher() {
...
+    Map<String, Object> req = getUserOrgUpdateRequest(false, false, false, false, false, false);
+    Request request = getRequest(false, false, true, req, ActorOperations.UPDATE_USER);
+    boolean result = testScenario(request, ResponseCode.errorUnsupportedField);
+    assertTrue(result);
+  }
+
+  @Test
+  public void testUpdateUserPrivateFailureWithPublic() {
```
Comment (amolnin) [inline service/src/test/java/org/sunbird/learner/actors/UserManagementActorTest.java]: testUpdateUserOrgFailureWithPublicApi

File: service/src/test/java/org/sunbird/learner/actors/UserManagementActorTest.java [hunk 3]
```diff
@@ -358,6 +365,157 @@ public void testUpdateUserFailureWithUserTypeTeacher() {
...
+  private void mockRequiredForUserOrgUpdate() {
+    when(ElasticSearchUtil.complexSearch(Mockito.any(), Mockito.anyString(), Mockito.anyString()))
+        .thenReturn(listOrg());
+    when(userService.getUserById(Mockito.anyString())).thenReturn(getUser(false));
+    when(cassandraOperation.insertRecord(
+            Mockito.anyString(), Mockito.anyString(), Mockito.anyMap()))
+        .thenReturn(null);
+  }
+
+  private Map<String, Object> listOrg() {
```
Comment (amolnin) [inline service/src/test/java/org/sunbird/learner/actors/UserManagementActorTest.java]: getListOrgResponse

File: service/src/test/java/org/sunbird/learner/actors/UserManagementActorTest.java [hunk 4]
```diff
@@ -358,6 +365,157 @@ public void testUpdateUserFailureWithUserTypeTeacher() {
...
+    Map<String, Object> req = getUserOrgUpdateRequest(true);
+    Request request = getRequest(false, false, true, req, ActorOperations.UPDATE_USER);
+    request.getContext().put(JsonKey.PRIVATE, true);
+    mockRequiredForUserOrgUpdate();
+    when(DataCacheHandler.getRoleMap()).thenReturn(roleMap(false));
+    boolean result = testScenario(request, ResponseCode.invalidRole);
+    assertTrue(result);
+  }
+
+  private void mockRequiredForUserOrgUpdate() {
```
Comment (amolnin) [inline service/src/test/java/org/sunbird/learner/actors/UserManagementActorTest.java]: mockForUserOrgUpdate

File: service/src/test/java/org/sunbird/learner/actors/UserManagementActorTest.java [hunk 5]
```diff
@@ -358,6 +365,157 @@ public void testUpdateUserFailureWithUserTypeTeacher() {
...
+    Map<String, Object> req = getUserOrgUpdateRequest(true);
+    ((Map) ((List) req.get(JsonKey.ORGANISATIONS)).get(0)).put(JsonKey.ORGANISATION_ID, "");
+    Request request = getRequest(false, false, true, req, ActorOperations.UPDATE_USER);
+    request.getContext().put(JsonKey.PRIVATE, true);
+    boolean result = testScenario(request, ResponseCode.mandatoryParamsMissing);
+    assertTrue(result);
+  }
+
+  @Test
+  public void testUpdateUserFailureWithInvalidRolesReqPrivateApi() {
```
Comment (amolnin) [inline service/src/test/java/org/sunbird/learner/actors/UserManagementActorTest.java]: InvalidRolesDataType

File: service/src/test/java/org/sunbird/learner/actors/UserManagementActorTest.java [hunk 6]
```diff
@@ -358,6 +365,157 @@ public void testUpdateUserFailureWithUserTypeTeacher() {
...
+  @Test
+  public void testUpdateUserFailureWithoutUserIdPrivateApi() {
+    Map<String, Object> req = getUserOrgUpdateRequest(false);
+    Request request = getRequest(false, false, true, req, ActorOperations.UPDATE_USER);
+    boolean result = testScenario(request, ResponseCode.errorUnsupportedField);
+    assertTrue(result);
+  }
+
+  @Test
+  public void testUpdateUserFailureWithPublicApi() {
```
Comment (amolnin) [inline service/src/test/java/org/sunbird/learner/actors/UserManagementActorTest.java]: all tests are UpdateUserOrg - not just update User

**Pull request merged**